shogun/utils/last_common_ancestor.py

```diff
@@ -31,9 +31,9 @@ def build_lca_map(gen: typing.Iterator, tree: Taxonomy) -> dict:
 
 
 def least_common_ancestor(taxa_set):
+    lca = []
     taxa_set = [_.split(';') for _ in taxa_set]
-    for i, level in enumerate(reversed(list(zip(*taxa_set)))):
-        if len(set(level)) == 1:
-            convert_i = lambda x: None if x == 0 else -x
-            return ';'.join([taxon_level[0] for taxon_level in list(zip(*taxa_set))[:convert_i(i)]])
-    return None
+    for level in zip(*taxa_set):
+        if len(set(level)) > 1 or any(_.endswith('__') for _ in level):
```

Comment:
@danknights I'm a bit concerned if this is the correct way we want to treat blanks, that is, taxonomy followed by two "__". Take the following code test example:

```python
    def test_blanks_kingdom_class(self):
        # normal scenario
        taxa = [
            'k__;p__Firmicutes;c__;o__Clostridiales;f__Clostridiaceae;g__Clostridium',
            'k__;p__Firmicutes;c__;o__Clostridiales;f__Clostridiaceae;g__Clostridium',
            'k__;p__Firmicutes;c__;o__Clostridiales;f__Clostridiaceae;g__Clostridium',
            'k__;p__Firmicutes;c__;o__Clostridiales;f__Clostridiaceae;g__Caminicella'
        ]
        obs = least_common_ancestor(taxa)
        exp = 'k__;p__Firmicutes;c__;o__Clostridiales;f__Clostridiaceae'
        # self.assertEqual(obs, exp)
        self.assertEquals(obs, None)
```
Should this result in `None` or the `exp =  'k__;p__Firmicutes;c__;o__Clostridiales;f__Clostridiaceae'`? I'm thinking `exp`.